build/main.rs

```diff
@@ -91,6 +91,124 @@ fn build_glue<P: AsRef<Path> + std::fmt::Debug>(include_path: &P) {
+    // It is possible though to use something like lazy_static! to create a new
+    // lua context and extract that information.
+    // For my (@wez) purposes, I actually don't want there to be a default path,
+    // so I'm just leaving this blank for the moment.
+    write!(glue, "pub const LUA_PATH_DEFAULT: &str = \"\";\n")?;
+    write!(glue, "pub const LUA_CPATH_DEFAULT: &str = \"\";\n")?;
+
+    write!(
+        glue,
+        "#[cfg(windows)] pub const LUA_DIRSEP: &str = \"\\\\\";\n"
+    )?;
+    write!(glue, "#[cfg(unix)] pub const LUA_DIRSEP: &str = \"/\";\n")?;
+
+    let pointer_bit_width: usize = env::var("CARGO_CFG_TARGET_POINTER_WIDTH")
+        .unwrap()
+        .parse()
+        .unwrap();
+    write!(
+        glue,
+        "pub const LUA_EXTRASPACE: c_int = {} / 8;\n",
+        pointer_bit_width
+    )?;
+
+    // This is generally hardcoded to this size
+    write!(glue, "pub const LUA_IDSIZE: c_int = 60;\n")?;
+
+    write!(glue, "pub const LUAL_BUFFERSIZE: c_int = 16 * ({} / 8) * std::mem::size_of::<LUA_NUMBER>() as c_int;\n", pointer_bit_width)?;
+
+    // Unless the target is restricted, the defaults are 64 bit
+    write!(glue, "pub type LUA_NUMBER = c_double;\n")?;
+    write!(glue, "pub type LUA_INTEGER = i64;\n")?;
+    write!(glue, "pub type LUA_UNSIGNED = u64;\n")?;
+
+    let version = if cfg!(feature = "luajit") || cfg!(feature = "lua51") {
+        // Note: luajit is ~lua 5.1, but it doesn't itself cross compile
+        (5, 1, 0)
+    } else if cfg!(feature = "lua52") {
+        (5, 2, 0)
+    } else if cfg!(feature = "lua53") {
+        (5, 3, 0)
+    } else if cfg!(feature = "lua54") {
+        (5, 4, 0)
+    } else {
+        unreachable!();
+    };
+
+    write!(
+        glue,
+        "pub const LUA_VERSION_NUM: c_int = {};\n",
+        (version.0 * 100) + version.1
+    )?;
+    write!(
+        glue,
+        "pub const LUA_VERSION: &str = \"Lua {}.{}\";\n",
+        version.0, version.1
+    )?;
+    write!(
+        glue,
+        "pub const LUA_RELEASE: &str = \"Lua {}.{}.{}\";\n",
+        version.0, version.1, version.2
+    )?;
+
+    let max_stack = if pointer_bit_width >= 32 {
+        1_000_000
+    } else {
+        15_000
+    };
+    write!(
+        glue,
+        "pub const LUA_REGISTRYINDEX: c_int = -{} - 1000;\n",
+        max_stack
+    )?;
+
+    // These two are only defined in lua 5.1
+    write!(glue, "pub const LUA_ENVIRONINDEX: c_int = -10001;\n")?;
+    write!(glue, "pub const LUA_GLOBALSINDEX: c_int = -10002;\n")?;
+
+    // This is only defined in lua 5.3 and up, but we can always generate its value here,
+    // even if we don't use it.
+    // This matches the default definition in lauxlib.h
+    write!(glue, "pub const LUAL_NUMSIZES: c_int = std::mem::size_of::<LUA_INTEGER>() as c_int * 16 + std::mem::size_of::<LUA_NUMBER>() as c_int;\n")?;
+
+    write!(
+        glue,
+        r#"
+pub const LUA_BITLIBNAME: &str = "bit32";
```

Comment:
In LuaJIT the library name is `bit` (see https://github.com/openresty/luajit2/blob/29a66f77d27ee669bc16e12f87ea312cd5376aab/src/lualib.h#L21)